### PR TITLE
Add a system for automatically cleaning up tables with player keys on disconnect

### PR DESCRIPTION
**Pull request by TomyLobo**
I went through these two:
```bash
git grep '\[ply\]' |grep -v self | less
git grep -i '\[[^]]*player[^]]*\]'
```

There are probably more tables with player keys, but those were the low-hanging fruits and should cover a lot of what we have.

### CODE REVIEW

**Comment by Divran** [comment]
Haven't tested ingame but it looks good.

**Comment by TomyLobo** [comment]
Meh, when was the last time that check succeeded?
I'm giving up, it's good enough. I even fixed some bugs along the way

**Comment by thegrb93** [comment]
Did you have to put the cleanup in the same pr >v<

**Comment by thegrb93** [comment]
The player table commits look fine. I didn't review all the cleanup changes yet, mind making a different branch and cherry picking them to that one and then removing them from this one?

File: lua/entities/gmod_wire_egp/lib/egplib/queuesystem.lua [hunk 1]
```diff
@@ -130,14 +130,17 @@ end
 timer.Create("EGP_Queue_Process", 1, 0, function()
 	local removetab = {}
 	for ply, tab in pairs(EGP.Queue) do
-		if !IsValid(ply) then removetab[ply] = true continue end
-		EGP:SendQueueItem(ply)
+		if not IsValid(ply) then
+			removetab[ply] = true
+		else
+			EGP:SendQueueItem(ply)
+		end
```
Comment (thegrb93) [inline lua/entities/gmod_wire_egp/lib/egplib/queuesystem.lua:137]: Going to swap this and remove the not

File: lua/wire/wire_paths.lua [hunk 2]
```diff
@@ -80,12 +80,15 @@ end
...
-		else
+		if not ply:IsValid() then
 			transmit_queues[ply] = nil
+		else
+			local nextinqueue = table.remove(queue, 1)
+			if nextinqueue then
+				TransmitPath(nextinqueue, ply)
+			else
+				transmit_queues[ply] = nil
+			end
```
Comment (thegrb93) [inline lua/wire/wire_paths.lua:91]: Flip this too

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
Everything else looks fine. Finally reviewed it all.

**Review by thegrb93 — APPROVED** [review]
Didn't see any errors. I don't have many EGP to test with though.

**Comment by TomyLobo** [comment]
Yeah i thought we had a requirement to fix all the luacheck errors that pop up.
That's why I put this in the same PR.
Thanks for testing it btw.

**Pull request merged**